microscopium/preprocess.py

```diff
@@ -592,8 +592,11 @@ def find_background_illumination(fns, radius=None, input_bitdepth=None,
 
     # return the median filter of that mean
     radius = radius or min(mean_image.shape) // 4
-    illum = ndi.percentile_filter(mean_image, percentile=(quantile * 100),
-                                  footprint=morphology.disk(radius))
+
+    # TODO: examine min & max, look at mean_image before and after img_as_uint, look at number of unique values
+    # TODO: want to see substantial range in the result
```

Comment:
The stretchlim below should be sufficient, so just remove these to-dos and we should be good to go!

... Except, the tests are now failing, so you'll need to check the test values and change the reference values accordingly... Let me know if you need help with this!

Reply:
`microscopium/tests/test_pre.py` (line 44)`illum_true` is the array you'll need to update after running the new and improved illumination correction function manually.

`microscopium/tests/test_pre.py`
Line 11: `def image_files()`
* generates the input image files (arrays) `i`, `j` and `k`

Line 41: `test_illumination_mean(image_files)`
* compares the result of the illumination correction function against the output array `illum_true`